### PR TITLE
add logging

### PR DESCRIPTION
**Pull request by karthik-dassana**

### CODE REVIEW

File: dassana/common.py [hunk 1]
```diff
@@ -147,8 +178,64 @@ def __init__(self, source, record_type, config_id, metadata=None, priority=None,
...
+            "crowdstrike_edr": "detection",
+            "crowdstrike_spotlight": "vulnerability",
+            "tenable_vulnerability": "vulnerability",
+            "snyk_vulnerability": "vulnerability",
+            "prisma_cloud_cspm": "cspm",
+            "prisma_cloud_cwpp": "vulnerability",
+            "prisma_cloud_security_group": "asset",
+            "prisma_cloud_instance": "asset",
+            "carbon_black_vulnerability": "vulnerability",
+            "ms_defender_endpoint_alert": "alert",
```
Comment (yatin-dassana) [inline dassana/common.py:195]: alert should change to detection

File: dassana/common.py [hunk 2]
```diff
@@ -147,8 +178,64 @@ def __init__(self, source, record_type, config_id, metadata=None, priority=None,
         self.ingestion_metadata = None
         self.custom_file_dict = dict()
         self.initialize_client()
+        self.report_state()
         self.file = open(self.file_path, 'a')
 
+    def report_state(self, status=None):
+        scope_id_mapping = {
```
Comment (yatin-dassana) [inline dassana/common.py:185]: missing: qualys, wiz

also this dict should be a global constant

File: dassana/common.py [hunk 3]
```diff
@@ -147,8 +178,64 @@ def __init__(self, source, record_type, config_id, metadata=None, priority=None,
...
+            logger.info(state_message)
+        else:
+            logger.error(state_message)
+
+        if dassana_partner:
+            if customer_status:
+                state_message["status"] = customer_status
+            if dassana_partner_client_id and dassana_partner_tenant_id:
+                state_message["siteId"] = dassana_partner_client_id
+                state_message["partnerTenantId"] = dassana_partner_tenant_id
```
Comment (yatin-dassana) [inline dassana/common.py:234]: this should be `tenantId`

File: dassana/common.py [hunk 4]
```diff
@@ -330,6 +417,7 @@ def cancel_job(self, exception_from_src):
 
         metadata = {"job_result": job_result_metadata}
         self.cancel_ingestion_job(metadata, "failed")
+        self.report_state(job_result_metadata["error_code"])
 
     def close(self, metadata=None):
```
Comment (yatin-dassana) [inline dassana/common.py:422]: missing `self.report_state` when job close

**Review by yatin-dassana — CHANGES_REQUESTED** [review]

File: dassana/dassana_logging.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+dassana_partner = get_partner()
+dassana_partner_client_id = get_partner_client_id()
+dassana_partner_tenant_id = get_partner_tenant_id()
+connection_id = get_ingestion_config_id()
+config_id = get_ingestion_config_id()
+is_logging_enabled = get_logging_enbled()
+
+topic_name = None
+if dassana_partner:
+    topic_name = dassana_partner + "_LOG_EVENT_TOPIC_NAME"
```
Comment (vinay-dassana) [inline dassana/dassana_logging.py:24]: `_LOG_EVENT_TOPIC_NAME` -> `_log_event_topic`

File: dassana/dassana_logging.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+        if state["status"] == "failed":
+            logger.error(msg=message["developerCtx"])
+        else:
+            logger.info(msg=message["developerCtx"])
+        
+        message = message["customerCtx"]
+        
+        if dassana_partner:
+            topic_name = dassana_partner+"_LOG_EVENT_TOPIC_NAME"
```
Comment (vinay-dassana) [inline dassana/dassana_logging.py:70]: reuse the var `topic_name`

File: dassana/dassana_env.py [hunk 7]
```diff
@@ -67,4 +67,31 @@ def is_internal_auth():
...
+    if "DASSANA_PARTNER_CLIENT_ID" not in os.environ:
+        return None
+    return str(os.environ["DASSANA_PARTNER_CLIENT_ID"])
+
+def get_partner_tenant_id():
+    if "DASSANA_PARTNER_TENANT_ID" not in os.environ:
+        return None
+    return str(os.environ["DASSANA_PARTNER_TENANT_ID"])
+    
+def get_logging_enbled():
```
Comment (vinay-dassana) [inline dassana/dassana_env.py:94]: remove this env var

File: dassana/common.py [hunk 8]
```diff
@@ -89,7 +92,17 @@ def get_ingestion_config(ingestion_config_id, app_id):
...
+    try:
+        response = response.json()
+        if not response.get("config"):
+            raise KeyError("config  missing in the response.")
+        else:
+            if not response["config"].get("_selectedScopeIds"):
+                raise KeyError("_selectedScopeIds are missing in the response.")            
+        return response
+    except:
+        raise KeyError()
```
Comment (vinay-dassana) [inline dassana/common.py:105]: do not raise any exception as KeyError

File: dassana/common.py [hunk 9]
```diff
@@ -112,6 +125,30 @@ def get_access_token():
                         is_internal=True)
     return response.json()["access_token"]
 
+def get_callback(
+    publish_future: pubsub_v1.publisher.futures.Future, data: str
+) -> Callable[[pubsub_v1.publisher.futures.Future], None]:
+    def callback(publish_future: pubsub_v1.publisher.futures.Future) -> None:
+        logger.info(f"Published Message with messageId: {publish_future.result(timeout=60)}")
```
Comment (vinay-dassana) [inline dassana/common.py:132]: log only on failure

File: dassana/common.py [hunk 10]
```diff
@@ -381,7 +420,7 @@ def get_ingestion_details(self):
                        is_internal=True)
         return res.json()
 
-    def cancel_ingestion_job(self, metadata, fail_type):
+    def cancel_ingestion_job(self, metadata, fail_type, exception_from_src=None):
```
Comment (vinay-dassana) [inline dassana/common.py:423]: remove exception from this method

File: dassana/common.py [hunk 11]
```diff
@@ -390,6 +429,7 @@ def cancel_ingestion_job(self, metadata, fail_type):
         logger.debug(f"Response Status: {res.status_code}")
         logger.debug(f"Request Body: {res.request.body}")
         logger.debug(f"Response Body: {res.text}")
+        log(self.source, fail_type, scope_id=self.metadata["scope"]["scopeId"], exception=exception_from_src)
```
Comment (vinay-dassana) [inline dassana/common.py:432]: move log to `cancel_job` method

**Review by vinay-dassana — CHANGES_REQUESTED** [review]

**Review by vinay-dassana — APPROVED** [review]

**Review by vinay-dassana — APPROVED** [review]

**Review by yatin-dassana — APPROVED** [review]

**Pull request merged**